test-clients/src/main/java/com/hedera/services/bdd/suites/token/TokenTransactSpecs.java

```diff
@@ -430,6 +446,189 @@ public HapiApiSpec balancesChangeOnTokenTransfer() {
 				);
 	}
 
+	public HapiApiSpec uniqueTokenTxnAccountBalance() {
+		return defaultHapiSpec("UniqueTokenTxnAccountBalance")
+				.given(
+						newKeyNamed("supplyKey"),
+						newKeyNamed("signingKeyTreasury"),
+						newKeyNamed("signingKeyFirstUser"),
+						cryptoCreate(FIRST_USER).key("signingKeyFirstUser"),
+						cryptoCreate(SECOND_USER),
+						cryptoCreate(TOKEN_TREASURY).key("signingKeyTreasury"),
+						tokenCreate(A_TOKEN)
+								.tokenType(TokenType.NON_FUNGIBLE_UNIQUE)
+								.supplyType(TokenSupplyType.INFINITE)
+								.initialSupply(0)
+								.supplyKey("supplyKey")
+								.treasury(TOKEN_TREASURY),
+						mintToken(A_TOKEN, List.of(ByteString.copyFromUtf8("memo"))),
+						tokenAssociate(FIRST_USER, A_TOKEN),
+						tokenAssociate(SECOND_USER, A_TOKEN)
+				).when(
+						cryptoTransfer(
+								movingUnique(1, A_TOKEN).between(TOKEN_TREASURY, FIRST_USER),
+								movingUnique(1, A_TOKEN).between(FIRST_USER, SECOND_USER)
+						).signedBy("signingKeyTreasury", "signingKeyFirstUser", DEFAULT_PAYER).via("cryptoTransferTxn")
+				).then(
+						getAccountBalance(TOKEN_TREASURY)
+								.hasTokenBalance(A_TOKEN, 0),
+						getAccountBalance(FIRST_USER)
+								.hasTokenBalance(A_TOKEN, 0),
+						getAccountBalance(SECOND_USER)
+								.hasTokenBalance(A_TOKEN, 1),
+
+						getTokenNftInfo(A_TOKEN, 1)
+								.hasSerialNum(1)
+								.hasMetadata(ByteString.copyFromUtf8("memo"))
+								.hasTokenID(A_TOKEN)
+								.hasAccountID(SECOND_USER),
+						getTxnRecord("cryptoTransferTxn").logged()
+				);
+	}
+
+	public HapiApiSpec uniqueTokenTxnWithNoAssociation() {
+		return defaultHapiSpec("UniqueTokenTxnWithNoAssociation")
+				.given(
+						cryptoCreate(TOKEN_TREASURY).balance(0L),
+						cryptoCreate(FIRST_USER).balance(0L),
+						newKeyNamed("supplyKey"),
+						tokenCreate(A_TOKEN)
+								.tokenType(TokenType.NON_FUNGIBLE_UNIQUE)
+								.supplyType(TokenSupplyType.INFINITE)
+								.initialSupply(0)
+								.supplyKey("supplyKey")
+								.treasury(TOKEN_TREASURY)
+				)
+				.when(
+						mintToken(A_TOKEN, List.of(ByteString.copyFromUtf8("memo")))
+				)
+				.then(
+						cryptoTransfer(
+								movingUnique(1, A_TOKEN).between(TOKEN_TREASURY, FIRST_USER)
+
+						).hasKnownStatus(FAIL_INVALID)
```

Comment:
The correct error for this test should be `TOKEN_NOT_ASSOCIATED_TO_ACCOUNT`. There must be something wrong

Reply:
It was a bug and corrected. 10x!